src/core/src/core_logic/PatchAssessor.py

```diff
@@ -77,11 +78,19 @@ def start_assessment(self):
 
     def telemetry_setup(self):
         """ Verifies if telemetry is available. Stops execution is not available """
+
+        if self.is_telemetry_setup:  # to avoid executing this again on implicit assessment i.e. 2nd assessment after install
+            return
+
         if self.execution_config.events_folder is None:
             error_msg = "The minimum Azure Linux Agent version prerequisite for Linux patching was not met. Please update the Azure Linux Agent on this machine."
             self.composite_logger.log_error(error_msg)
             raise Exception(error_msg)
 
-        #ToDo: Ensure telemetry is setup correctly at this point
         self.composite_logger.log("The minimum Azure Linux Agent version prerequisite for Linux patching was met.")
+        self.telemetry_writer.events_folder_path = self.execution_config.events_folder
+        self.telemetry_writer.set_operation_id(self.execution_config.activity_id)
+        self.telemetry_writer.telemetry_startup()
+        self.is_telemetry_setup = True
```

Comment:
To much tight coupling going on here (actually the whole function). The innards of telemetry_writer are way too exposed here.